### PR TITLE
ARROW-11360: [Rust] [DataFusion] Improve CSV "No files found" error message

### PR DESCRIPTION
**Pull request by jameswinegar**
Helps with the debugging process.

### CODE REVIEW

File: rust/datafusion/src/datasource/csv.rs [hunk 1]
```diff
@@ -66,7 +66,10 @@ impl CsvFile {
                 let mut filenames: Vec<String> = vec![];
                 common::build_file_list(path, &mut filenames, options.file_extension)?;
                 if filenames.is_empty() {
-                    return Err(DataFusionError::Plan("No files found".to_string()));
+                    return Err(DataFusionError::Plan(format!(
+                        "No files found at {path}",
```
Comment (andygrove) [inline rust/datafusion/src/datasource/csv.rs:70]: Can we show the file extension it was looking for as well?

File: rust/arrow-flight/src/arrow.flight.protocol.rs [hunk 2]
```diff
@@ -498,9 +498,8 @@ pub mod flight_service_server {
     #[async_trait]
     pub trait FlightService: Send + Sync + 'static {
         #[doc = "Server streaming response type for the Handshake method."]
-        type HandshakeStream: Stream<
-            Item = Result<super::HandshakeResponse, tonic::Status>,
-        > + Send
+        type HandshakeStream: Stream<Item = Result<super::HandshakeResponse, tonic::Status>>
+            + Send
```
Comment (andygrove) [inline rust/arrow-flight/src/arrow.flight.protocol.rs:502]: This formatting change shouldn't be part of this PR.

Comment (jameswinegar) [reply rust/arrow-flight/src/arrow.flight.protocol.rs]: fixed

File: rust/datafusion/src/datasource/csv.rs (hunk 1 above)
Comment (jameswinegar) [reply rust/datafusion/src/datasource/csv.rs]: fixed

**Comment by alamb** [comment]
Hi @jameswinegar  -- it appears that you branch got messed up due to the arrow release process (which involves force pushing to `apache/master`). Can you please rebase your changes against the current apache master branch? Sorry -- the process is being updated to avoid force pushes to master in subsequent releases

**Comment by jameswinegar** [comment]
> Hi @jameswinegar -- it appears that you branch got messed up due to the arrow release process (which involves force pushing to `apache/master`). Can you please rebase your changes against the current apache master branch? Sorry -- the process is being updated to avoid force pushes to master in subsequent releases

Should be fixed, let me know if this works?

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Looks good to me -- Thanks @jameswinegar !

**Review by andygrove — APPROVED** [review]
Thanks @jameswinegar LGTM
